src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2885,7 +2885,10 @@ internal override AnalyzerDriver AnalyzerForLanguage(ImmutableArray<DiagnosticAn
 
         internal void SymbolDeclaredEvent(Symbol symbol)
         {
-            if (EventQueue != null) EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(this, symbol));
+            if (EventQueue?.IsCompleted == false)
+            {
+                EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(this, symbol));
```

Comment:
Can we assert that this is in fact an error case? For example, we can verify that the symbol is not reachable by descending from source module's global namespace.